### PR TITLE
feat: azure workload identity support

### LINKED ISSUE #4487: Azure workload Identity support
it is require to add Azure workload Identity support.

- The main use-case  aks in cluster scanning

### PR DESCRIPTION
**Pull request by chen-keinan**
## Description
azure workload identity support

## Related issues
- Close #4487
- Related https://github.com/aquasecurity/trivy-operator/issues/1057
## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).

### CODE REVIEW

File: pkg/fanal/image/registry/azure/azure.go [hunk 1]
```diff
@@ -24,13 +35,26 @@ func (r *Registry) CheckOptions(domain string, _ types.RegistryOptions) error {
...
-		return "", "", xerrors.Errorf("ACR credential error: %w", err)
+		return "", "", xerrors.Errorf("unable to get an access token: %w", err)
 	}
-	token, err := credStore.Get(ctx, r.domain)
+	rt, err := refreshToken(ctx, aadToken.Token, r.domain)
 	if err != nil {
-		return "", "", xerrors.Errorf("unable to get a token: %w", err)
+		return "", "", xerrors.Errorf("unable to refresh token: %w", err)
+	}
+	return "00000000-0000-0000-0000-000000000000", *rt.RefreshToken, err
```
Comment (knqyf263) [inline pkg/fanal/image/registry/azure/azure.go:50]: It returns a refresh token. Can it be used for registry authentication? In general, refresh tokens are used for renewing access tokens. I just want to make sure of it.

File: pkg/fanal/image/registry/azure/azure.go [hunk 2]
```diff
@@ -24,13 +35,26 @@ func (r *Registry) CheckOptions(domain string, _ types.RegistryOptions) error {
...
+}
+
+func refreshToken(ctx context.Context, accessToken string, domain string) (containerregistry.RefreshToken, error) {
+	tenantID := os.Getenv("AZURE_TENANT_ID")
```
Comment (knqyf263) [inline pkg/fanal/image/registry/azure/azure.go:54]: Does it mean only `AZURE_TENANT_ID` must be set? How about other variables such as `AZURE_CLIENT_ID`? Please update the document if needed.
https://aquasecurity.github.io/trivy/v0.41/docs/advanced/private-registries/acr/

File: pkg/fanal/image/registry/azure/azure.go (hunk 1 above)
Comment (chen-keinan) [reply pkg/fanal/image/registry/azure/azure.go:50]: trivy can't use `AAD token` for ACR authentication,  the `GetFromExchange()` method exchange `AAD token` with `ACR token` , see grant type is `access_token`.

AAD token will fail authentication for ACR. t is compatible with [prev. Implementation](https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/image/registry/azure/auth.go#L96)

File: pkg/fanal/image/registry/azure/azure.go (hunk 2 above)
Comment (chen-keinan) [reply pkg/fanal/image/registry/azure/azure.go:54]: yes  all three env. variables must be set as describe in existing doc.
```
# must set TRIVY_USERNAME empty char
export AZURE_CLIENT_ID$(echo $SP_DATA | jq -r .appId)
export AZURE_CLIENT_SECRET$(echo $SP_DATA | jq -r .password)
export AZURE_TENANT_ID$(echo $SP_DATA | jq -r .tenant)
```

File: pkg/fanal/image/registry/azure/azure.go (hunk 1 above)
Comment (knqyf263) [reply pkg/fanal/image/registry/azure/azure.go:50]: Thanks for your explanation!

**Review by knqyf263 — APPROVED** [review]

File: pkg/fanal/image/registry/azure/azure.go (hunk 2 above)
Comment (KyrreHaugland) [reply pkg/fanal/image/registry/azure/azure.go:54]: How is it possible to use the workload identity within the trivy-operator helm chart?

Comment (chen-keinan) [reply pkg/fanal/image/registry/azure/azure.go:54]: @KyrreHaugland follow this issue https://github.com/aquasecurity/trivy-operator/issues/1057 , I'll add support next week in operator

**Pull request merged**